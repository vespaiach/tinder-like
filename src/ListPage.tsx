import { Avatar, List, ListItem, ListItemAvatar, ListItemText } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { UserProfile } from './types';

interface ListPageProps {
    profiles: UserProfile[];
}

const useStyles = makeStyles((theme) => ({
    listRoot: {},
}));

export default function ListPage({ profiles }: ListPageProps) {
    const classes = useStyles();
    return (
        <List className={classes.listRoot}>
            {profiles.map((p) => (
                <ListItem key={p.id} disableGutters>
                    <ListItemAvatar>
                        <Avatar alt={p.name} src={p.picture} />
                    </ListItemAvatar>
                    <ListItemText
                        primary={p.name}
                        secondary={
                            <span>
                                Age: {p.age} | Gender: {p.gender}
                            </span>
                        }
                    />
                </ListItem>
            ))}
        </List>
    );
}
