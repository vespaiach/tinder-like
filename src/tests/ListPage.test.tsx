import { render } from '@testing-library/react';

import ListPage from '../ListPage';

describe('Test ListPage component', () => {
    test('should render empty list', () => {
        const { container } = render(<ListPage profiles={[]} />);
        expect(container).toMatchSnapshot();
    });
    test('should render one item', () => {
        const { container } = render(
            <ListPage
                profiles={[
                    {
                        id: '1',
                        name: 'Toan Nguyen',
                        picture: 'picture',
                        age: 1,
                        gender: 'male',
                    },
                ]}
            />
        );
        expect(container).toMatchSnapshot();
    });
});
