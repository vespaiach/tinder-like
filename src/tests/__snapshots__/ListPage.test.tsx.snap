// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ListPage component should render empty list 1`] = `
<div>
  <ul
    class="MuiList-root makeStyles-listRoot-1 MuiList-padding"
  />
</div>
`;

exports[`Test ListPage component should render one item 1`] = `
<div>
  <ul
    class="MuiList-root makeStyles-listRoot-2 MuiList-padding"
  >
    <li
      class="MuiListItem-root"
    >
      <div
        class="MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            alt="Toan Nguyen"
            class="MuiAvatar-img"
            src="picture"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Toan Nguyen
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          <span>
            Age: 
            1
             | Gender: 
            male
          </span>
        </p>
      </div>
    </li>
  </ul>
</div>
`;
